src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationAndGeneratorDriverTranslationAction_Actions.cs

```diff
@@ -28,6 +25,17 @@ public override Task<Compilation> TransformCompilationAsync(Compilation oldCompi
                     return UpdateDocumentInCompilationAsync(oldCompilation, _oldState, _newState, cancellationToken);
                 }
 
+                private static async Task<Compilation> UpdateDocumentInCompilationAsync(
+                    Compilation compilation,
+                    DocumentState oldDocument,
+                    DocumentState newDocument,
+                    CancellationToken cancellationToken)
+                {
+                    return compilation.ReplaceSyntaxTree(
+                        await oldDocument.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false),
+                        await newDocument.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false));
+                }
```

Comment:
this helper was up in SolutionState i believe.  i just moved it here since this is the only place that uses it.

Reply:
Frankly we should just inline it...there's more lines of parameters than there are actual code. :smile: